dotCMS/src/main/java/com/dotcms/rest/api/v1/apps/AppsHelper.java

```diff
@@ -164,24 +176,65 @@ Response getAppSiteView(
             final AppDescriptor appDescriptor = appDescriptorOptional.get();
             final Host host = hostAPI.find(siteId, user, false);
             if (null == host) {
-                throw new DotDataException(
-                        String.format(" Couldn't find any host with identifier `%s` ", siteId));
+                throw new DoesNotExistException(
+                      String.format(" Couldn't find any host with identifier `%s` ", siteId)
+                );
             }
 
             final Optional<AppSecrets> optionalAppSecrets = appsAPI
                     .getSecrets(key, true, host, user);
-            if (optionalAppSecrets.isPresent()) {
-                final AppSecrets appSecrets = protectHiddenSecrets(
-                        optionalAppSecrets.get());
-                final SiteView siteView = new SiteView(host.getIdentifier(), host.getHostname(),
-                        appSecrets.getSecrets());
-                return Optional.of(new AppView(appDescriptor, 1L,
-                        ImmutableList.of(siteView)));
-            }
+
+                //We need to return a view with all the secrets and also descriptors of the remaining parameters merged.
+                //So we're gonna need a copy of the params on the yml.
+                final Map<String, ParamDescriptor> descriptorParams = new HashMap<>(appDescriptor.getParams());
```

Comment:
maybe it should go to a private method so this comment can go in the method javadoc

Reply:
I think it's fine